File: LLM_Translation/C_programs/brotli.c
#define BROTLI_TRUE 1
#define BROTLI_FALSE 0


static int ParseInt(const char* s, int low, int high, int* result) {
  int value = 0;
  int i;
  for (i = 0; i < 5; ++i) {
    char c = s[i];
    if (c == 0) break;
    if (s[i] < '0' || s[i] > '9') return BROTLI_FALSE;
    value = (10 * value) + (c - '0');
  }
  if (i == 0) return BROTLI_FALSE;
  if (i > 1 && s[0] == '0') return BROTLI_FALSE;
  if (s[i] != 0) return BROTLI_FALSE;
  if (value < low || value > high) return BROTLI_FALSE;
  *result = value;
  return BROTLI_TRUE;
}

int main(){
  
}

File: LLM_Translation/Rust_programs_compiled/brotli.rs
const BROTLI_TRUE: i32 = 1;
const BROTLI_FALSE: i32 = 0;

fn parse_int(s: &str, low: i32, high: i32) -> Option<i32> {
    let mut value = 0;
    let mut chars = s.chars();
    
    for i in 0..5 {
        if let Some(c) = chars.next() {
            if c < '0' || c > '9' {
                return None;
            }
            value = (10 * value) + (c as i32 - '0' as i32);
        } else {
            break;
        }
    }
    
    if s.len() == 0 || (s.len() > 1 && s.starts_with('0')) || value < low || value > high {
        return None;
    }
    
    Some(value)
}

fn main() {
    // The main function is currently empty.
    // You can add code here to test the `parse_int` function.
}